orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTask.groovy

```diff
@@ -188,10 +182,40 @@ class CreateBakeTask implements RetryableTask {
       requestMap.remove("baseName")
     }
 
+    if (stage.context.baseOs == null) {
+      requestMap.custom_managed_image_name = stage.context.managedImage as String
+      if (stage.context.managedImage as String == null) {
+        requestMap.sku = stage.context.sku as String
+        requestMap.offer = stage.context.offer as String
+        requestMap.publisher = stage.context.publisher as String
+      }
+    }
+
     def request = mapper.convertValue(requestMap, BakeRequest)
     if (!bakery.config.roscoApisEnabled) {
       request.other().clear()
     }
+
     return request
   }
+
+  private static PackageType getCustomPackageType(StageExecution stage) {
+    PackageType type
+    if(stage.context.osType != "linux") {
+      type = PackageType.NUPKG
```

Comment:
I would return instead the type. by doing this you don't need the `else` clause. you can directly do 
```
type = stage.context.packageType as PackageType
      if (type == null) {
        type = new OperatingSystem(stage.context.managedImage as String).getPackageType()
      }
```
without the else clause. For readability but it's just a personal preference.

Reply:
great catch, will do the changes